pkg/expapi/validation/validation.go

```diff
@@ -53,6 +60,67 @@ func validateHorizontalPodAutoscalerSpec(autoscaler expapi.HorizontalPodAutoscal
 	return allErrs
 }
 
+func ValidatePositiveIntOrPercent(intOrPercent util.IntOrString, fieldName string) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if intOrPercent.Kind == util.IntstrString {
+		if !util.IsValidPercent(intOrPercent.StrVal) {
```

Comment:
I was thinking about that and decided to leave it.
Users specifying a surge of 200% while increasing replicas from 10 to 20 seemed fine to me.
I am validating that its > 0.

But yes for maxUnavailable, we can check that its not more than 100%. Will add that. Thanks!